core/extensions/impl/memory.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "extensions/impl/memory.hpp"
+
+namespace kagome::extensions {
+
+  Memory::Memory() {
+    offset_ = 0;
+  }
+
+  Memory::Memory(size_t size) : Memory() {
+    memory_.resize(size);
+  }
+
+  void Memory::resize(size_t newSize) {
+    // Ensure the smallest allocation is large enough that most allocators
+    // will provide page-aligned storage. This hopefully allows the
+    // interpreter's memory to be as aligned as the memory being simulated,
+    // ensuring that the performance doesn't needlessly degrade.
+    //
+    // The code is optimistic this will work until WG21's p0035r0 happens.
```

Comment:
is this copied from somewhere? what is WG21 p0035r0 ?

Reply:
Yes, from binaryen